parla/parray/core.py

```diff
@@ -60,7 +60,12 @@ def _to_current_gpu(self) -> None:
         """
         Move the array to current GPU, do nothing if already on the device.
         """
-        self.array = cupy.asarray(self.array)  # asarray by default copy to current device
+        if isinstance(self.array, numpy.ndarray):
```

Comment:
I see. It will be merged after your PR is merged. Do you agree with the below update calling copy_from_device_async and do your PR reflect this? The `asarray()` was problematic. As I reported before, it didn't work when I used on scheduler.

Reply:
Yineng, let me merge this to dev-scheduler branch first. Let me handle your suggestion when we merge this onto dev branch (after your branch is done)

Reply:
I think my branch is already done? I am working on find grained movement now.

Reply:
I think that is also an important choice. Should PArray be thread-safe? Or do the upper layers who will use PArray support thread-safe? I prefer the latter one. I am not for sure if this discussion is about threading. Before doing that, could you discuss with other people first?